### PR TITLE
optionally define config variables via environment variables

### PR DESCRIPTION
**Pull request by sambvfx**
Make the use of the config.py file optional if not using environment variables.

### CODE REVIEW

File: zync.py [hunk 1]
```diff
@@ -28,11 +28,15 @@ class ZyncConnectionError(Exception):
 class ZyncPreflightError(Exception):
     pass
 
-current_dir = os.path.dirname(os.path.abspath(__file__)) 
-config_path = os.path.join(current_dir, 'config.py')
-if not os.path.exists(config_path):
-  raise ZyncError('Could not locate config.py, please create.')
-from config import *
+if os.environ.get('ZYNC_URL'):
```
Comment (jurekPapiorek) [inline zync.py:31]: Hi Sam,

I'm working right now on a PR that allows passing zync_url to the constructor of zync.Zync. Would that work for you? My PR should go live next week.

**Comment by sambvfx** [comment]
That's ideal actually and seems like the proper way to handle it. Interestingly, I'll probably still store the values as environment variables and use them when constructing that class within our wrapper. I also understand your need to provide simple configurations to the not-so-tech savvy.

**Comment by bcipriano** [comment]
Hi Sam - we've deployed the change that does this to both `master` and `release` branches. Please feel free to start using that code.

**Comment by sambvfx** [comment]
Hi Brian,

This still requires the `config.py` file to be present (and have the `ZYNC_URL`) variable to even import the `zync` module. If you'd like to maintain the `config.py` sidecar file for providing defaults, I would prefer something like the following:

_sudo code_

```
class Zync(HTTPBackend):
    def __init__(self, script_name, token, timeout=30.0, application=None, 
                 disable_ssl_certificate_validation=False, url=None):
        if url is None:
            url = self.get_url()

    def get_url():
        from config import ZYNC_URL
        return ZYNC_URL
```

The default behavior could be easily overridden in a subclass.

The other zync modules have similar problems.

Cheers,
Sam

**Comment by broniek** [comment]
Friendly ping :)

**Comment by sambvfx** [comment]
Hi all,

Thanks for implementing the ability to construct the Zync object with a URL. However I'm still having to maintain my own copy of the repo because of what I mentioned before:

> This still requires the config.py file to be present (and have the ZYNC_URL) variable to even import the zync module.

I'd like to just use your repo as-is and not have to make changes to the files you distribute.

**Comment by bcipriano** [comment]
Hi Sam -

Sorry for the delay. These changes (including the `zync-maya` and `zync-nuke` changes) all look good to me. I'll merge these into our master branch so you can start using that.

After a day or two the changes should also make their way into the `release` branch as well, which is what is installed when you run our Plugin Installer.

Thanks!!

**Comment by sambvfx** [comment]
Thanks Brian!

**Pull request merged**